src/components/Markets/index.tsx

```diff
@@ -123,52 +65,85 @@ export class Markets extends React.Component<MarketsProps, MarketsState> {
         });
 
         return <span className={className}>{cell}</span>;
-    }
+    }, []);
 
-    private mapRows = (cell: CellData) => {
+    const mapRows = React.useCallback((cell: CellData) => {
         const isChangeValue = typeof(cell) === 'string' && (cell.charAt(0) === '-' || cell.charAt(0) === '+');
 
-        return isChangeValue ? this.renderChange(cell as string) : cell;
+        return isChangeValue ? renderChange(cell as string) : cell;
+    }, [renderChange]);
+
+    const filterType = (headerKey: string, key: string) => (item: CellData[]) => {
+        const typeIndex = props.headers ? props.headers.indexOf(headerKey) : defaultHeaders.indexOf(headerKey);
+
+        return (item[typeIndex] as string).includes(key);
     };
 
-    private filterType = (headerKey: string, searchKey: string) => (item: CellData[]) => {
-        const typeIndex = this.props.headers ? this.props.headers.indexOf(headerKey) : this.defaultHeaders.indexOf(headerKey);
+    const createUniqueCurrencies = (currencies: string[], market: string) => {
+        const isCurrencyUnique = (currency: string) => !currencies.includes(currency);
+
+        const marketCurrencies = market.split('/').map((c: string) => c.trim());
+        const uniqueCurrencies = marketCurrencies.filter(isCurrencyUnique);
 
-        return (item[typeIndex] as string).includes(searchKey);
+        return currencies.concat(uniqueCurrencies);
     };
 
-    private get filters() {
-        const { data } = this.props;
+    const transformCurrencyToFilter = (currency: string) => ({
+        name: currency,
+        filter: filterType('Pair', currency),
+    });
+
+    const filtersFn = () => {
```

Comment:
try to find better name for this function

Reply:
Previously, it was a getter for filters, right now it is a function